arch/rv32i/src/support.rs

```diff
@@ -0,0 +1,27 @@
+use core::ops::FnOnce;
+
+#[inline(always)]
+/// NOP instruction
+pub fn nop() {
+    unsafe {
+        asm!("nop" :::: "volatile");
+    }
+}
+
+#[inline(always)]
+/// WFI instruction
+pub unsafe fn wfi() {
+    asm!("wfi" :::: "volatile");
+}
+
+/// TODO: implement
+pub unsafe fn atomic<F, R>(f: F) -> R
+where
+    F: FnOnce() -> R,
+{
```

Comment:
You can make the body of this `unimplemented!()`, which the compiler is aware of and can then warn folks if this gets used

Reply:
Well it is called, but just doesn't really protect against anything.